content/posts/gsoc-physiopy/index.md

```diff
@@ -2,6 +2,104 @@
 title = 'GSoC Physiopy'
 date = 2024-09-03T19:33:41+03:00
 draft = false
+toc = true
 +++
 
-Google Summer of Code
\ No newline at end of file
+During the summer of 2024, I had the chance to become a Google Summer of Code student, contributing in the [Physiopy](https://physiopy.github.io/) community, under the [INCF](https://www.incf.org/) organization. In this page, which is also serving as my Work Product Submission for finalizing my GSoC participation, the project will be presented, how it progressed during the summer and what future directions it could take. 
+Emphasis is going to be given to the development breakdown, the rationale of all the contributions and the main ideas for future development.
+
+## 1. Project Overview
+{{< lead >}}
+*"Semi-Automated Workflows for Physiological Signals"*
+{{< /lead >}}
+
+In general, the project entailed the upgrade and restructuring of existing Physiopy packages, `peakdet` and `phys2denoise`, in order to be used more harmoniously with each other. 
+Following the upgrade, the development of semi-automated workflows was planned, along with the accompanying CLI. 
+
+## 2. Pull Request Synopsis
+### `peakdet`
+- https://github.com/physiopy/peakdet/pull/67
+- https://github.com/physiopy/peakdet/pull/62
+### `phys2denoise`
+- https://github.com/physiopy/phys2denoise/pull/54
+- https://github.com/physiopy/phys2denoise/pull/56
+- https://github.com/physiopy/phys2denoise/pull/57
+### `physutils`
+- https://github.com/physiopy/physutils/pull/2
+- https://github.com/physiopy/physutils/pull/4
+- https://github.com/physiopy/physutils/pull/7
+
+## 3. Development Overview
+
+### 3.1 Infrastructure 
```

Comment:
Are you planning to comment on loguru here? This ended up being quite significant for the project overall, especially since it will be important to document some of the loguru/pydra challenges and how it's affecting the current workflow in this report.